README.md

```diff
@@ -38,13 +41,156 @@ class Translations {
+      defaultLocale: 'en',
+      supportedLocales: ['en', 'fr'],
+      // if you are using flavors, you'll need to specify the default and current one
+      defaultFlavor: 'default',
+      currentFlavor: 'flavor1',
+      // This method is called to load the default ARB files, easiest way is to load them from assets
+      dataLoader: (locale, flavor) async {
+        print('local load $locale and $flavor');
+        if (flavor == 'default') {
+          return await rootBundle.loadString('assets/arb/translations_$locale.arb');
+        }
+        return await rootBundle.loadString('assets/arb/translations_${flavor}_$locale.arb');
+      },
+      // If you want to download your ARB files from a remote server
+      // You need to specfiy a custom data loader like this
+      updateDataLoader: (locale, flavor) async {
+        print('Remote load $locale and $flavor');
+        if (locale == 'en' && flavor == 'flavor1') {
+            await Future.delayed(Duration(seconds: 10));//simulate some slow network response
+            return await rootBundle.loadString('assets/arb/translations_remote_$locale.arb');
+        }
+        return null;  // no update
+      },
+      // Once your translations are updated from your remote files this callback will be triggered, you'll need to rebuild in order to see the changes
+      onTranslationsUpdated: () {
+        print('TX updated, need rebuild');
+        setState(() {});
+      },
+      // Builder to build your custom class containing your translations
+      translationsBuilder: () => Translations(),
+    ),
+]
+```
 
-Flavor ARB doesn't have to contain all the sentences, if a sentence is not present in the flavor ARB, the default sentence from the locale will be used.
+Here is a little more details of the `TranslationsDelegate` parameters:
 
-### Configuration
+- `defaultLocale`: default locale to use for your app.
+- `defaultFlavor`: optional default flavor to use for your app, default to 'default'.
+- `currentFlavor`: optional current flavor to use for your app, default to 'default'.
+- `overrideCurrentLocale`: optional locale to use for your app, it will override the system locale.
+- `dataLoader`: callback to get the ARB files for a given locale and flavor, this is call by jimtl to let you provide the ARB files when needed.
+- `updateDataLoader`: optional, to keep your translations up to date from a remote server, this is the callback you need. It should returned the ARB content for a given locale and flavor or null if no update is needed.
+- `onTranslationsUpdated`: optional, once your translations have been updated remotely by `updateDataLoader` this callback is called for your to rebuild your widget and see the changes.
+- `supportedLocales`: supported locales for your app.
+- `translationsBuilder`: builder to provide your custom class containing your sentences.
+
+### Flavors specificities
+
+Flavor support are optional in this package, but it's here in case you ever need it :)
```

Comment:
Flavor support is optional in this package, but it's here in case you ever need it :)